blacktie/tx/src/main/cpp/HttpTxManager.cxx

```diff
@@ -207,7 +207,7 @@ bool HttpTxManager::recover(XAWrapper *resource)
 	if (ri.status_code == 200) {
 		LOG4CXX_DEBUG(httptxlogger, "recovery: told TM about participant "
 			<< resource->get_name() << " xid: " << xid);
-
+		_branches.erase(resource->get_name());
```

Comment:
Well spotted.

I think we should also be removing the branch after a successful commit or abort. Do you agree?